src/System Application/App/AppSource Gallery/src/AppSourceProductManager.Codeunit.al

```diff
@@ -15,100 +15,117 @@ using System.RestClient;
 
+    var
+        AppSourceJsonUtilities: Codeunit "AppSource Json Utilities";
+        AppSourceProductManagerDependencies: Interface "AppSource Product Manager Dependencies";
+        IsDependenciesIntrfaceSet: boolean;
+        CatalogProductsUriLbl: label 'https://catalogapi.azure.com/products', Locked = true;
+        CatalogApiVersionQueryParamNameLbl: label 'api-version', Locked = true;
+        CatalogApiVersionQueryParamValueLbl: label '2023-05-01-preview', Locked = true;
+        CatalogApiOrderByQueryParamNameLbl: label '$orderby', Locked = true;
+        CatalogMarketQueryParamNameLbl: label 'market', Locked = true;
+        CatalogLanguageQueryParamNameLbl: label 'language', Locked = true;
+        CatalogApiFilterQueryParamNameLbl: Label '$filter', Locked = true;
+        CatalogApiSelectQueryParamNameLbl: Label '$select', Locked = true;
+        AppSourceListingUriLbl: Label 'https://appsource.microsoft.com/%1/product/dynamics-365-business-central/%2', Comment = '%1=Language ID, such as en-US, %2=Url Query Content', Locked = true;
+        AppSourceUriLbl: Label 'https://appsource.microsoft.com/%1/marketplace/apps?product=dynamics-365-business-central', Comment = '1%=Language ID, such as en-US', Locked = true;
+        NotSupportedOnPremisesErrorLbl: Label 'Not supported on premises.';
+        UnsupportedLanguageNotificationLbl: Label 'Language %1 is not supported by AppSource. Defaulting to "en". Change the language in the user profile to use another language.', Comment = '%1=Language ID, such as en';
+        UnsupportedMarketNotificationLbl: Label 'Market %1 is not supported by AppSource. Defaulting to "us". Change the region in the user profile to use another market.', Comment = '%1=Market ID, such as "us"';
+
     #region Dependency Interface implementation
-    procedure AzureADTenant_GetCountryLetterCode(): Text[2]
+    procedure GetCountryLetterCode(): Code[2]
     var
-        entraTenant: Codeunit "Azure AD Tenant";
+        EntraTenant: Codeunit "Azure AD Tenant";
     begin
-        exit(entraTenant.GetCountryLetterCode());
+        exit(EntraTenant.GetCountryLetterCode());
     end;
 
-    procedure AzureAdTenant_GetPreferredLanguage(): Text
+    procedure GetPreferredLanguage(): Text
     var
-        entraTenant: Codeunit "Azure AD Tenant";
+        EntraTenant: Codeunit "Azure AD Tenant";
     begin
-        exit(entraTenant.GetPreferredLanguage());
+        exit(EntraTenant.GetPreferredLanguage());
     end;
 
-    procedure AzureADTenant_GetAadTenantID(): Text
+    procedure GetAadTenantID(): Text
     var
-        entraTenant: Codeunit "Azure AD Tenant";
+        EntraTenant: Codeunit "Azure AD Tenant";
     begin
-        exit(entraTenant.GetAadTenantID());
+        exit(EntraTenant.GetAadTenantID());
     end;
 
-    procedure AzureKeyVault_GetAzureKeyVaultSecret(SecretName: Text; var Secret: SecretText);
+    procedure GetAzureKeyVaultSecret(SecretName: Text; var Secret: SecretText);
```

Comment:
This seems like a bad idea... 'Internal' is not secure and this method allows retrieval of any secret... Why do you need this helper method? 
and if you need it, it should be marked as [Scope('OnPrem')]

Reply:
@WaelAbuSeada wanna take a look?

Reply:
Good point about onprem scope. The current dependency management requires an implementation of the keyvault dependency.